src/gui/sdlmain.cpp

```diff
@@ -1473,40 +1492,28 @@ static void GUI_StartUp(Section * sec) {
 #endif
 	/* Get Keyboard state of numlock and capslock */
 	SDL_Keymod keystate = SDL_GetModState();
-	if(keystate&KMOD_NUM) startup_state_numlock = true;
-	if(keystate&KMOD_CAPS) startup_state_capslock = true;
-}
-
-void Mouse_AutoLock(bool enable) {
-	sdl.mouse.autolock=enable;
-	if (sdl.mouse.autoenable) sdl.mouse.requestlock=enable;
-	else {
-		SDL_ShowCursor(enable?SDL_DISABLE:SDL_ENABLE);
-		sdl.mouse.requestlock=false;
-	}
+	if (keystate&KMOD_NUM) startup_state_numlock = true;
+	if (keystate&KMOD_CAPS) startup_state_capslock = true;
 }
 
 static void HandleMouseMotion(SDL_MouseMotionEvent * motion) {
-	if (sdl.mouse.locked || !sdl.mouse.autoenable)
+	if (mouse_is_captured)
 		Mouse_CursorMoved((float)motion->xrel*sdl.mouse.xsensitivity/100.0f,
 						  (float)motion->yrel*sdl.mouse.ysensitivity/100.0f,
 						  (float)(motion->x-sdl.clip.x)/(sdl.clip.w-1)*sdl.mouse.xsensitivity/100.0f,
 						  (float)(motion->y-sdl.clip.y)/(sdl.clip.h-1)*sdl.mouse.ysensitivity/100.0f,
-						  sdl.mouse.locked);
+						  mouse_is_captured);
 }
 
 static void HandleMouseButton(SDL_MouseButtonEvent * button) {
 	switch (button->state) {
 	case SDL_PRESSED:
-		if (sdl.mouse.requestlock && !sdl.mouse.locked) {
-			GFX_CaptureMouse();
+		if ((sdl.mouse.capture_choice != MouseCaptureType::Never && !mouse_is_captured)
+		    || (button->button == SDL_BUTTON_MIDDLE && mouse_is_captured)) {
```

Comment:
I'm afraid some games might use middle mouse button… I thought Settlers is the one, but that one uses left+right "special" click: d88f6db3b9193d276756b01703c6e9822e8646e5. I googled a bit and the only one I could find was a [user-created mod for Dune II](https://forum.dune2k.com/topic/27527-dune-ii-mouse-helper-utility-for-improving-dos-version-controls/), but I think we need to do some digging.

And I believe macOS does not support middle mouse button out of the box and requires a separate app installed? (form over function, I guess)

Reply:
MacOS honors middle click (and wheel) if the USB device provides it.  Just need to plug in a PC mouse.  Behaviour is also confirmed on a VM that provides 3-button passthrough (like VirtualBox).

But yeah - I didn't realize DOS games recognized the middle mouse button! By the time I got a 3-button mouse, I had moved onto Windows 95.  Turns our the early 90s Sierra VGA point and click games would rotate the action cursor (walk/talk/look/...) on a middle press.  Who know!  

So I've made this configurable and updated the PR message. 

Edit: tested on all three platforms now.